lib/Doctrine/DBAL/Types/SimpleArrayType.php

```diff
@@ -44,9 +44,8 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
      */
     public function convertToDatabaseValue($value, AbstractPlatform $platform)
     {
-        if (!$value) {
-            return null;
-        }
+        if ($value === null) return null;
+        if (!$value) return '';
```

Comment:
@zeroedin-bill no because `if (!$value)` check will also be true for an empty array and thus returning `null` (currently).
But unfortunately I think this is a BC break because users are relying on the current behaviour I suppose. We had similar discussion with `JsonArray` and could agree on a change because of keeping BC (I know this sounds weird but IMO we cannot change the assumption that an empty array will evaluate to `null`). Existing code with checks against `null` will break with this PR....